### DISCUSSION TITLE
Table Row count: JS for formula fields & binding. Please help

### DISCUSSION THREAD

**Discussion by sergey-dme** [Budibase/budibase#3627] [Help] [answered]
Hi BB Team,

First of all, congratulations on BB v1.0x release! 🎉 

I'm trying to use the following JS in Conditions for Table property - Row count. The idea is changing a number of table rows displayed depending on client/user screen height. (Some my users have screens with resolution 1080px, some of them more than 1080 and if I use fixed table rows it is not good, as a screen space are not used efficiently - too much empty space remaining on screens with big resolution. In the same time I don't want to get a h-scroll on low res screens). To solve this case I used:

```
var hscr = window.innerHeight;
return hscr;
```
But it always returns `0`! I did not find here on BB GitHub that `window` functions were block due to security or something. It should work. And..  `document.documentElement.clientHeight` does not work as well. Also returns `0`

Please help. Thank you

**Comment by mjashanks** [comment, marked as answer]
Hey Sergey.

Javascript is sandboxed. This means that it does not have access to the DOM. 

The purpose of Javascript in Budibase is to perform logic that returns values for bindings - not DOM manipulations.

**Reply by sergey-dme** [reply]
Hey @mjashanks thanks for replying. Then may be the team would consider to add an option dynamic table row count, when it dynamically inc/dec. depending on the container or parent object height
